services/src/main/java/org/exoplatform/officeonline/documents/OfficeOnlineNewDocumentTemplatePlugin.java

```diff
@@ -0,0 +1,93 @@
+
+import org.exoplatform.container.ExoContainerContext;
+import org.exoplatform.container.component.BaseComponentPlugin;
+import org.exoplatform.container.xml.InitParams;
+import org.exoplatform.container.xml.ObjectParameter;
+import org.exoplatform.ecm.webui.component.explorer.documents.DocumentTemplate;
+import org.exoplatform.ecm.webui.component.explorer.documents.NewDocumentService;
+import org.exoplatform.ecm.webui.component.explorer.documents.NewDocumentServiceImpl;
+import org.exoplatform.ecm.webui.component.explorer.documents.NewDocumentTemplatePlugin;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+
+/**
+ * The Class OnlyOfficeNewDocumentTemplatePlugin.
+ */
+public class OfficeOnlineNewDocumentTemplatePlugin extends BaseComponentPlugin implements NewDocumentTemplatePlugin {
+
+  /** The Constant LOG. */
+  protected static final Log       LOG                              =
+                                       ExoLogger.getLogger(OfficeOnlineNewDocumentTemplatePlugin.class);
+
+  /**   The DOCUMENT_TYPES_CONFIGURATION param. */
+  private static final String      DOCUMENT_TEMPLATES_CONFIGURATION = "document-templates-configuration";
+
+  /** The document types. */
+  protected List<DocumentTemplate> templates                        = Collections.emptyList();
+
+  /** The provider. */
+  protected String                 provider;
+
+  /**
+   * Instantiates a new new document type plugin.
+   *
+   * @param initParams the init params
+   */
+  public OfficeOnlineNewDocumentTemplatePlugin(InitParams initParams) {
+    ObjectParameter typesParam = initParams.getObjectParam(DOCUMENT_TEMPLATES_CONFIGURATION);
+    if (typesParam != null) {
+      Object obj = typesParam.getObject();
+      if (obj != null && NewDocumentServiceImpl.DocumentTemplatesConfig.class.isAssignableFrom(obj.getClass())) {
+        NewDocumentServiceImpl.DocumentTemplatesConfig config = NewDocumentServiceImpl.DocumentTemplatesConfig.class.cast(obj);
+        this.templates = config.getTemplates();
+        this.provider = config.getProvider();
+      } else {
+        LOG.error("The document templates are not set");
+      }
+    }
+  }
+
+  /**
+   * Gets the provider.
+   *
+   * @return the provider
+   */
+  @Override
+  public String getProvider() {
+    return provider;
+  }
+
+  /**
+   * Gets the templates.
+   *
+   * @return the templates
+   */
+  @Override
+  public List<DocumentTemplate> getTemplates() {
+    return templates;
+  }
+
+  /**
+   * Creates the document.
+   *
+   * @param parent the parent
+   * @param title the title
+   * @param template the template
+   * @return the node
+   * @throws Exception the exception
+   */
+  @Override
+  public Node createDocument(Node parent, String title, DocumentTemplate template) throws Exception {
+    LOG.debug("Creating new document {} from template {}", title, template);
+    NewDocumentService documentService = ExoContainerContext.getCurrentContainer()
```

Comment:
NewDocumentService can be injected via constructor dependency by the container.

Reply:
Fixed